service/worker/scheduler/workflow.go

```diff
@@ -1337,6 +1337,7 @@ func (s *scheduler) startWorkflow(
 			Header:                   newWorkflow.Header,
 			LastCompletionResult:     lastCompletionResult,
 			ContinuedFailure:         continuedFailure,
+			UserMetadata:             newWorkflow.UserMetadata,
```

Comment:
I'm assuming local activity inputs are not required to match for determinism, right?

Reply:
Yes. Dug a bit around and also confirmed with @cretz . This change should be good.